### ISSUE TITLE
Review plotly renderers

### ISSUE THREAD

**Issue by Juanlu001** [poliastro/poliastro#710]
See https://github.com/plotly/plotly.py/pull/1474#issuecomment-483843514

**Comment by Juanlu001** [comment]
Encouraged by [the release notes of the upcoming Plotly 4.0](https://community.plot.ly/t/introducing-plotly-py-4-0-0rc1/25639), I tried Spyder and, unfortunately, an appropriate renderer is not selected automatically. I then tried to install orca, but it was clashing with another binary with the same name in my operative system, so I gave up.

**Comment by Juanlu001** [comment]
I insisted a bit more in installing orca, and manually selecting the renderer worked:

![Screenshot from 2019-07-14 19-01-36](https://user-images.githubusercontent.com/316517/61186810-f2651b00-a669-11e9-864f-8fabd0d51c4f.png)

**Comment by Juanlu001** [comment]
I have been exploring this topic a little bit and commented in several issues along the way:

* https://github.com/plotly/plotly.py/issues/1664
* https://github.com/plotly/orca/issues/238
* https://github.com/jupyter/nbconvert/issues/944#issuecomment-511216610

**Comment by Juanlu001** [comment]
From [this comment](https://github.com/jupyter/nbconvert/issues/944#issuecomment-511218705) and the evidence I gathered, *if the latest combination of nbconvert, sphinx-rtd-theme and plotly works out of the box*, we should probably switch to using `Figure` instead of `FigureWidget` here:

https://github.com/poliastro/poliastro/blob/1ebe69f133c46c26e4f9d3a6ad1955b0fcb8f4b9/src/poliastro/plotting/_base.py#L27

Advantages:

* `Figure` works out of the box in HTML and slides (hopefully in docs too!)
* We do not need `ipywidgets` in the default installation

Non-advantages:

* `Figure` still requires `orca` + renderer tuning to display in Spyder

**Comment by jonmmease** [comment]
Hi @Juanlu001,

Thanks for trying things out!  After reading your observations, I took another look at Spyder, and it looks like we can use environment variables to auto-detect that plotly.py is running inside Spyder.  Here's the commit where I just added this feature https://github.com/plotly/plotly.py/commit/b39db734cfd8ddfa1f84dd83fe1f5eb59b21904b to the `v4_integration` branch (which will be merged into master with the final v4 release).

Basically, if we see the `SPYDER_ARGS` environment variable *and* orca is installed successfully then auto-enable the `svg` renderer. Thanks!

**Comment by Juanlu001** [comment]
That's awesome, thanks a lot @jonmmease! Still figuring out what did I break with the plots in nbsphinx (a long struggle!), hope I can narrow down the issue this week.

**Comment by Juanlu001** [comment]
Seems like sphinx-rtd-theme has again some issues with plotly: https://github.com/readthedocs/sphinx_rtd_theme/issues/788

**Comment by Juanlu001** [comment]
Also, as can be seen here https://rtd-js-test.readthedocs.io/en/alabaster/ the plot shows with/without `.show()`, and regardless of the renderer chosen (only tried ones that do not need orca installed). I guess `notebook_connected` would be better to save some space in the notebook, but for today I focused on diagnosing the issue.

**Comment by Juanlu001** [comment]
The renderers are reviewed and the notebooks are now working with `notebook_connected`, so we can close this!
